### PR TITLE
Multiserver: Only update client status for a slot when the first enters and the last leaves

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
Turn off the lights on your way out. Also updated the `hint_cost` on the network doc since it was wrong.

## How was this tested?
Connected with my game and tested with each combination and state, with temporary logging.

## If this makes graphical changes, please attach screenshots.
Screenshot_8
Screenshot_9
Screenshot_10

### CODE REVIEW

**Comment by alwaysintreble** [comment]
ClientStatus could likely be exposed in a different way and done later/separately if i'm the only one who actually wants this, but I think it gets us closer to the Ready/Lobby system. I only tested that this works with the .net lib so should probably check it doesn't break any clients.

**Comment by Jarno458** [comment]
I don't there is a much point in putting the values in the NetworkPlayer's array unless you are also going to send updated versions of that array each time a client status changes, like when they connect or disconnect or change it programmatically.
As right now clients are only informed about the NetworkPlayers's once during the connected packet and everytime time someone changes their alias.

I can see some usage of being able to know what players are playing or have completed their goals, but it will likely be a lot of updates in a big game like a1000p async.

**Comment by alwaysintreble** [comment]
My idea for using this slot is that you would specifically request it and not have it auto updated. Maybe we put it in as part of the datapackage so we can SetNotify it?

File: MultiServer.py [hunk 1]
```diff
@@ -746,7 +746,8 @@ async def on_client_disconnected(ctx: Context, client: Client):
 
 
 async def on_client_joined(ctx: Context, client: Client):
-    update_client_status(ctx, client, ClientStatus.CLIENT_CONNECTED)
+    if ctx.client_game_state[client.team, client.slot] not in {ClientStatus.CLIENT_READY, ClientStatus.CLIENT_PLAYING}:
```
Comment (Berserker66) [inline MultiServer.py:749]: I'd prefer if this operated on numerical values instead, such that if the status is not higher, raise it.

Comment (Berserker66) [reply MultiServer.py:749]: the background is that it's possible more states are added later and then it's easy to miss updating lists like the above.

Comment (alwaysintreble) [reply MultiServer.py]: I opted to check that the status is only "not connected" so if newer status get added it doesn't matter, and still not using some arbitrary value.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**